spec/unit/rich_text/rendering/embedded_form_renderer_spec.rb

```diff
@@ -0,0 +1,71 @@
+require "formalist/rich_text/rendering/embedded_form_renderer"
+
+RSpec.describe Formalist::RichText::Rendering::EmbeddedFormRenderer do
+  subject(:renderer) { described_class.new(container, namespace: namespace, paths: paths) }
+
+  let(:container) {{
+    "article" => "top_level_article",
+    "embedded_forms.article" => "namespaced_article",
+    "embedded_forms.newsletter.article" => "newsletter_article",
+    "embedded_forms.newsletter.components.article" => "newsletter_components_article",
+    "embedded_forms.general.article" => "general_article",
+   }}
+
+  context "no namespace or paths configured" do
+    let(:namespace) { nil }
+    let(:paths) { [] }
+
+    describe "#resolve_key" do
+      it "returns the top level result" do
+        expect(renderer.send(:resolve_key, "article")).to eq "article"
```

Comment:
Rather than testing a private method, would it be worth testing via `EmbeddedFormRenderer#call` instead? 🤔 We could register callable procs in the container and create expectations around their outputs? Something like: 

```ruby
  let(:container) {{
    "article" => -> (*_) { "top_level_article" },
    "embedded_forms.article" => -> (*_) { "namespaced_article" },
    "embedded_forms.newsletter.article" => -> (*_) { "newsletter_article" },
    "embedded_forms.newsletter.components.article" => -> (*_) { "newsletter_components_article" },
    "embedded_forms.general.article" => -> (*_) { "general_article" },
   }}
```